invisible_cities/cities/beersheba.py

```diff
@@ -311,6 +312,14 @@ def write_deconv(df):
     return write_deconv
 
 
+def kdst_writer(h5out):
+    """ This function protects for kdst of None type"""
+    def writer(kdst):
+        if isinstance(kdst, pd.DataFrame): kdst_from_df_writer(h5out)(kdst)
```

Comment:
As standing now, `kdst_from_df_writer(h5out)` would be executed for each element of the source, shouldn't it happen only once?

Reply:
Right, I would go for a filter based solution as you suggested above